### ISSUE TITLE
add test for driver tablet routing logic

### ISSUE THREAD

**Issue by denesb** [scylladb/scylladb#16855]
Now that the python driver included in the frozen toolchain, supports tablets, we should add tests for this.

**Comment by denesb** [comment]
/cc @bhalevy 

@avikivity what tests you had in mind exactly? Checking that the driver sends the requests to the correct node:shard?

**Comment by avikivity** [comment]
Checking that the payload is there when we miss the tablet, and matches the tablets table.

Checking that the payload is not there when we hit the tablet.

**Comment by denesb** [comment]
Who should work on this? @sylwiaszunejko or somebody from my team (maybe me)?

**Comment by mykaul** [comment]
@dkropachev  ?

**Comment by dkropachev** [comment]
@avikivity , do you want these tests in driver repo or in core ?

**Comment by nyh** [comment]
> @avikivity , do you want these tests in driver repo or in core ?

This is a good question. It depends what you **really** want to test:

1. If the goal is to test some core Scylla feature like tablets or that requests are answer properly with tablets, and so on, it should be in the Scylla repo, where the Python driver is only considered a way to exercise a specific feature and not the thing being tested. In this case, these tests should be skipped when not running  a driver able to run this test (see an example driver_bug_1() in test/cql-pytest) - I don't want to force developers to install any specific version of the Cassandra driver (let alone insist on installing a very recent Scylla ones) just to run tests. In cql-pytest, the Python driver is just a tool - it's not the thing being tested.

2. If the goal is to test how the driver sends requests to the correct nodes or tablets or whatever, these tests should be in the driver repo. I think that here to you'll need to consider how to skip those tests when running against CQL implementations which aren't the most recent Scylla - after all, we probably want to run these driver tests against Cassandra too (or do we? I think we should but I don't know if you do.).

**Comment by dkropachev** [comment]
> > @avikivity , do you want these tests in driver repo or in core ?
> 
> This is a good question. It depends what you **really** want to test:
> 
> 1. If the goal is to test some core Scylla feature like tablets or that requests are answer properly with tablets, and so on, it should be in the Scylla repo, where the Python driver is only considered a way to exercise a specific feature and not the thing being tested. In this case, these tests should be skipped when not running  a driver able to run this test (see an example driver_bug_1() in test/cql-pytest) - I don't want to force developers to install any specific version of the Cassandra driver (let alone insist on installing a very recent Scylla ones) just to run tests. In cql-pytest, the Python driver is just a tool - it's not the thing being tested.
> 2. If the goal is to test how the driver sends requests to the correct nodes or tablets or whatever, these tests should be in the driver repo. I think that here to you'll need to consider how to skip those tests when running against CQL implementations which aren't the most recent Scylla - after all, we probably want to run these driver tests against Cassandra too (or do we? I think we should but I don't know if you do.).

Exactly, so my assumption it should be done in core repos, @avikivity , could you please confirm that?

**Comment by swasik** [comment]
> Exactly, so my assumption it should be done in core repos, [@avikivity](https://github.com/avikivity) , could you please confirm that?

@avikivity friendly ping :-)

**Comment by avikivity** [comment]
The tests in this repo should test the server side, the tests in the driver repos should test the driver side.

**Comment by dkropachev** [comment]
> The tests in this repo should test the server side, the tests in the driver repos should test the driver side.

Makes sense, my assumption is this what needs to be tested:
1. Test that certain replies recieve `TABLETS_ROUTING_V1` extension if tablets are enabled and do not receive if it disabled.
2. Test that certain replies recieve `tablets-routing-v1` custom payload with tablets in it.

@mykaul, given last @avikivity reply I am pretty sure this issue is for core dev and have nothing to do with @scylladb/drivers-team, am I wrong ?

**Comment by mykaul** [comment]
> > The tests in this repo should test the server side, the tests in the driver repos should test the driver side.
> 
> Makes sense, my assumption is this what needs to be tested:
> 
> 1. Test that certain replies recieve `TABLETS_ROUTING_V1` extension if tablets are enabled and do not receive if it disabled.
> 2. Test that certain replies recieve `tablets-routing-v1` custom payload with tablets in it.
> 
> [@mykaul](https://github.com/mykaul), given last [@avikivity](https://github.com/avikivity) reply I am pretty sure this issue is for core dev and have nothing to do with [@scylladb/drivers-team](https://github.com/orgs/scylladb/teams/drivers-team), am I wrong ?

I think it was the drivers team who implemented the core side as well (but I may be wrong).

**Comment by avikivity** [comment]
@sylwiaszunejko implemented driver tablet routing in core

**Comment by dkropachev** [comment]
> > [@sylwiaszunejko](https://github.com/sylwiaszunejko) implemented driver tablet routing in core
> 
> [@dkropachev](https://github.com/dkropachev) so is it fine to keep it in your team?

Sure, we will keep it on us.

**Comment by dani-tweig** [comment]
Closing this issue as it was moved to Jira.
Please continue the thread in https://scylladb.atlassian.net/browse/SCYLLADB-1964
